tests/components/mill/test_init.py

```diff
@@ -0,0 +1,126 @@
+"""Tests for Mill init."""
+
+from unittest.mock import patch
+
+import pytest
+
+from homeassistant.components import mill
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from tests.common import MockConfigEntry, mock_coro
+
+
+async def test_setup_with_cloud_config(hass):
+    """Test setup of cloud config."""
+    mock_entry = MockConfigEntry(
+        domain=mill.DOMAIN,
+        data={
+            mill.CONF_USERNAME: "user",
+            mill.CONF_PASSWORD: "pswd",
+            mill.CONNECTION_TYPE: mill.CLOUD,
+        },
+    )
+    mock_entry.add_to_hass(hass)
+    with patch("mill.Mill.fetch_heater_and_sensor_data", return_value={}) as mock_fetch:
+        with patch("mill.Mill.connect", return_value=True) as mock_connect:
+            assert await mill.async_setup_entry(hass, mock_entry)
```

Comment:
Style nit: Perhaps combine to a single with like the local test does.
here and below.